<launch>
    <arg name="machine" default=""/>

    <rosparam file="$(find ar3_hardware_interface)/config/controllers.yaml" command="load"/>
    <rosparam file="$(find ar3_hardware_interface)/config/joint_offsets.yaml" command="load"/>
    <rosparam file="$(find ar3_moveit_config)/config/joint_limits.yaml" command="load" ns="ar3"/>
    <rosparam file="$(find marsha_hardware)/config/stepper_calibration.yaml" command="load"/>

    <!-- Load universal robot description format (URDF) -->
    <!-- Could be done in ar3_moveit_bringup launch file -->
    <param name="robot_description" command="$(find xacro)/xacro $(find ar3_description)/urdf/ar3_hardware.urdf.xacro"/>
    <!-- Loaded in planning_context.launch
        <param name="robot_description_semantic" textfile="$(find ar3_moveit_config)/config/ar3.srdf" />-->

    <include file="$(find marsha_hardware)/launch/hardware_comm.launch">
        <arg name="machine" value="$(arg machine)"/>
    </include>

    <node name="gripper_driver" pkg="marsha_core" type="motor.py" machine="$(arg machine)"/>

    <node name="marsha_hardware_interface" pkg="marsha_hardware" type="marsha_hardware_interface_node" output="screen" machine="$(arg machine)"/>

    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/" 
        args="/ar3/controllers/state
              /ar3/controllers/position"
        machine="$(arg machine)"/>
        <!--/ar3/controllers/effort-->

</launch>